packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import { createContext, useEffect, useMemo, useState } from "react";
+
+import { AuthProvider, type AuthProviderParams, useAuth as useAuthCore } from "@crossmint/client-sdk-auth-core";
+import { EVMSmartWallet, SmartWalletSDK } from "@crossmint/client-sdk-smart-wallet";
+
+type AuthWalletProviderParams = AuthProviderParams & {
+    embeddedWallets: {
+        type: "evm-smart-wallet";
+        defaultChain: "polygon-amoy" | "base-sepolia";
+        createOnLogin: "all-users" | "off";
+    };
+};
+
+export const WalletContext = createContext<{ wallet: EVMSmartWallet | null }>({
+    wallet: null,
+});
+
+export function CrossmintAuthProvider(props: AuthWalletProviderParams) {
+    return (
+        <AuthProvider {...props}>
+            <AuthWalletProvider {...props}></AuthWalletProvider>
+        </AuthProvider>
+    );
+}
+
+function AuthWalletProvider(props: AuthWalletProviderParams) {
+    const { jwt } = useAuthCore();
+
+    const [wallet, setWallet] = useState<EVMSmartWallet | null>(null);
+
+    const smartWalletSDK = useMemo(() => SmartWalletSDK.init({ clientApiKey: props.apiKey }), [props.apiKey]);
+
+    const getOrCreateWallet = async (jwt: string) => {
+        try {
+            const wallet = await smartWalletSDK.getOrCreateWallet({ jwt }, props.embeddedWallets.defaultChain);
+            setWallet(wallet);
+        } catch (e: any) {
+            console.error("There was an error creating a wallet ", e);
+            throw e;
+        }
```

Comment:
How do errors like this typically get exposed to external devs? How would an external dev catch and handle this?

Reply:
Good point, this runs only in the useEffect :thinking:
Maybe we just console.error for now and figure out error handling later. Wdy think?

Reply:
sgtm